### ISSUE TITLE
ResolveBufferDescriptors not handling rank-reducing subviews blocks e2e mmt4d tests

### ISSUE THREAD

**Issue by bjacob** [iree-org/iree#10176]
I tried testing vmvx/mmt4d for float32 only as that's supposed to be supported. But e2e matmul tests involve testcases where the K-dimension is 1, and some drop-unit-dims patterns kick in and drop the 1-size dimension and that results in rank-reducing subviews.

rank-reducing subviews are currently explicitly not supported in vmvx:
https://sourcegraph.com/github.com/iree-org/iree@e4dc88c3d1f6bd7d88cf418e775e549d4e8b6e37/-/blob/compiler/src/iree/compiler/Dialect/VMVX/Transforms/ResolveBufferDescriptors.cpp?L36

Testcase `x.mlir`  (just the relevant testcase from the generated e2e matmul tests, as-is):

```
func.func @matmul_10x1xf32_times_1x10xf32_into_10x10xf32(%lhs: tensor<10x1xf32>, %rhs: tensor<1x10xf32>, %acc: tensor<10x10xf32>) -> tensor<10x10xf32> {
  %result = linalg.matmul ins(%lhs, %rhs: tensor<10x1xf32>, tensor<1x10xf32>) outs(%acc: tensor<10x10xf32>) -> tensor<10x10xf32>
  return %result: tensor<10x10xf32>
}
```

Repro:

```
tools/iree-compile --output-format=vm-bytecode --mlir-disable-threading --mlir-print-ir-after-all --iree-hal-target-backends=vmvx --iree-vmvx-enable-microkernels "--iree-flow-mmt4d-target-options=enable_generic_slow arch=x86_64" ~/x.mlir -o /tmp/x.vmfb
```

Result:

```
// -----// IR Dump After FoldTensorExtractOp (iree-codegen-fold-tensor-extract-op) //----- //
module {
  func.func @matmul_10x1xf32_times_1x10xf32_into_10x10xf32_dispatch_0() {
    %c0 = arith.constant 0 : index
    %c1 = arith.constant 1 : index
    %0 = hal.interface.binding.subspan set(0) binding(0) type(storage_buffer) offset(%c0) alignment(64) : memref<10xf32>
    memref.assume_alignment %0, 64 : memref<10xf32>
    %1 = hal.interface.binding.subspan set(0) binding(1) type(storage_buffer) offset(%c0) alignment(64) : memref<16x2xf32>
    memref.assume_alignment %1, 64 : memref<16x2xf32>
    %2 = memref.subview %1[0, 0] [10, 1] [1, 1] : memref<16x2xf32> to memref<10xf32, affine_map<(d0) -> (d0 * 2)>>
    %workgroup_id_x = hal.interface.workgroup.id[0] : index
    %3 = affine.apply affine_map<()[s0] -> (s0 * 2)>()[%workgroup_id_x]
    %4 = memref.subview %0[%3] [2] [1] : memref<10xf32> to memref<2xf32, affine_map<(d0)[s0] -> (d0 + s0)>>
    %base_buffer, %offset, %sizes, %strides = vmvx.get_buffer_descriptor %4 : memref<2xf32, affine_map<(d0)[s0] -> (d0 + s0)>> -> !util.buffer, index, index, index
    %5 = memref.subview %2[%3] [2] [1] : memref<10xf32, affine_map<(d0) -> (d0 * 2)>> to memref<2xf32, affine_map<(d0)[s0] -> (d0 * 2 + s0)>>
    %base_buffer_0, %offset_1, %sizes_2, %strides_3 = vmvx.get_buffer_descriptor %5 : memref<2xf32, affine_map<(d0)[s0] -> (d0 * 2 + s0)>> -> !util.buffer, index, index, index
    vmvx.copy in(%base_buffer offset %offset strides[%c0, %strides] : !util.buffer) out(%base_buffer_0 offset %offset_1 strides[%c0, %strides_3] : !util.buffer) sizes(%c1, %sizes_2) : f32
    return
  }
}

/usr/local/google/home/benoitjacob/x.mlir:3:13: error: Unable to resolve all strided buffer descriptors:
```

In GDB I could see that the issue was we were hitting the above `"rank reducing subview not supported"` case on this one:

```
    %2 = memref.subview %1[0, 0] [10, 1] [1, 1] : memref<16x2xf32> to memref<10xf32, affine_map<(d0) -> (d0 * 2)>>
```

@stellaraccident I can try to tackle this if you can describe what needs to be done?

**Comment by bjacob** [comment]
This blocks #9904

**Comment by alliemartens** [comment]
@MaheshRavishankar @bjacob Can you help to give this a priority?

**Comment by stellaraccident** [comment]
There's not a fundamental reason for the restriction: was just missing a test see and not wanting to guess.

**Comment by bjacob** [comment]
FIxed by #10192

**Comment by MaheshRavishankar** [comment]
I guess we can give it P0 and take credit for closing a P0 bug :P
